evm/src/main/java/org/hyperledger/besu/evm/operation/SelfDestructOperation.java

```diff
@@ -60,19 +75,22 @@ public OperationResult execute(final MessageFrame frame, final EVM evm) {
     final Address address = frame.getRecipientAddress();
     final MutableAccount account = frame.getWorldUpdater().getAccount(address).getMutable();
 
-    frame.addSelfDestruct(address);
+    if (!eip6780Semantics || frame.wasCreatedInTransaction(address)) {
+      frame.addSelfDestruct(address);
+    }
 
     final MutableAccount recipient =
         frame.getWorldUpdater().getOrCreate(recipientAddress).getMutable();
 
-    if (!account.getAddress().equals(recipient.getAddress())) {
-      recipient.incrementBalance(account.getBalance());
+    Wei contractBalance = account.getBalance();
+    if (eip6780Semantics || !account.getAddress().equals(recipient.getAddress())) {
+      recipient.incrementBalance(contractBalance);
```

Comment:
With the renames, this statement becomes a lot clearer IMO...

`!addressOfAccountToMaybeDestruct.equals(selfDestructBeneficiary.getAddress())) {
      selfDestructBeneficiary.incrementBalance(balanceOfDestructedAccount);`